Apps/AdminWebClient/src/Server/Services/AuthenticationService.cs

```diff
@@ -123,35 +123,37 @@ public void SetLastLoginDateTime()
         {
             ClaimsPrincipal? user = this.httpContextAccessor.HttpContext?.User;
             AuthenticationData authData = this.GetAuthenticationData();
-            if (authData.IsAuthenticated && authData.User != null && user != null)
+            if (authData is not { IsAuthenticated: true, User: not null } || user == null)
```

Comment:
As bad as it gets - expressions like that are just not readable.

Reply:
The double negative from inverting the if statement is the main culprit I think.  Luckily, this is in AdminWebClient and not long for this world.